x/evm/keeper/chainKeeper.go

```diff
@@ -239,6 +239,17 @@ func (k chainKeeper) GetBurnerAddressAndSalt(ctx sdk.Context, token types.ERC20T
 	salt := types.Hash(common.BytesToHash(crypto.Keccak256Hash(bz).Bytes()))
 
 	var initCodeHash types.Hash
+
+	// always use the latest burner byte code for external token
+	if token.IsExternal() {
+		burnerCode, ok := k.GetBurnerByteCode(ctx)
+		if !ok {
+			return types.Address{}, types.Hash{}, fmt.Errorf("burner code not found for chain %s", k.chainLowerKey)
+		}
+		initCodeHash = types.Hash(crypto.Keccak256Hash(burnerCode))
+		return types.Address(crypto.CreateAddress2(gatewayAddr, salt, initCodeHash.Bytes())), salt, nil
+	}
+
```

Comment:
```suggestion
	var tokenBurnerCodeHash string
	if token.IsExternal() {
		// always use the latest burner byte code for external token
		burnerCode, ok := k.GetBurnerByteCode(ctx)
		if !ok {
			return types.Address{}, types.Hash{}, fmt.Errorf("burner code not found for chain %s", k.chainLowerKey)
		}

		tokenBurnerCodeHash = crypto.Keccak256Hash(burnerCode).Hex()
	} else {
		tokenBurnerCodeHash = token.GetBurnerCodeHash().Hex()
	}
```
so that the following constructor code is re-used.